### PR TITLE
Fix exception message for invalid custom string plugin

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

When you pass a wrong custom plugin nickname:

```python
import pytorch_lightning as pl

pl.Trainer(
    fast_dev_run=True,
    gpus=0,
    num_processes=1,
    distributed_backend='ddp_cpu',
    plugins='invalid',
)
```
you get:

```
pytorch_lightning.utilities.exceptions.MisconfigurationException:  invalid is not a supported lightning custom plugin. If you're trying to pass a custom plugin, please pass this as an object to Trainer(plugins=[MyPlugin()]. Supported plugins as string input: <generator object PluginConnector._convert_str_to_plugin.<locals>.<genexpr> at 0x7fae1dd189e0>.
```

This fixes the code so that we get:

```
Supported plugins as string input: ['ddp_sharded', 'native_amp', 'apex_amp']
```

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/plugin_connector.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def _convert_str_to_plugin(self, plugin):
                     f" {plugin} is not a supported lightning custom plugin."
                     " If you're trying to pass a custom plugin, please pass this as an object to"
                     " Trainer(plugins=[MyPlugin()]."
-                    f" Supported plugins as string input: {(e.name for e in LightningCustomPlugins)}."
+                    f" Supported plugins as string input: {([e.name for e in LightningCustomPlugins])}."
```
Comment (rohitgr7) [inline pytorch_lightning/plugins/plugin_connector.py:123]:
```suggestion
                    f" Supported plugins as string input: {[e.name for e in LightningCustomPlugins]}."
```

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/plugins/plugin_connector.py (hunk 1 above)
Comment (SeanNaren) [reply pytorch_lightning/plugins/plugin_connector.py:123]: Thanks! i thought it was a bit strange haha

**Review by ydcjeff — APPROVED** [review]

**Review by Borda — APPROVED** [review]
LGTM

**Review by awaelchli — APPROVED** [review]
